src/smt/qi_queue.cpp

```diff
@@ -188,7 +188,7 @@ namespace smt {
 
     void qi_queue::display_instance_profile(fingerprint * f, quantifier * q, unsigned num_bindings, enode * const * bindings, unsigned proof_id, unsigned generation) {
         if (m.has_trace_stream()) {
-            m.trace_stream() << "[instance] 0x0";
+            m.trace_stream() << "[instance] " << f->get_data_hash();
```

Comment:
f can be null.
I removed references to f because the format of printing the pointers was platform specific. 
Can you make this safer by using get_data_hash(f) instead, and 

unsigned get_data_hash(fingerprint* f) {
      return f ? f->get_data_hash() : 0;
}